src/app/courses/courses.component.ts

```diff
@@ -23,20 +23,43 @@ export class CoursesComponent implements OnInit {
     public orderByName: OrderByPipe,
     public filterSearch: FilterSearchPipe
   ) {
-    this.courses = this.coursesService.getAll(this.page, this.displayLimit);
+    var _this = this;
+    this.coursesService.getCoursesFromServer().subscribe({
+      next(data: any) {
+        if (data) {
+          _this.coursesService.setList(data);
+          _this.courses = _this.coursesService.getAll(_this.page, _this.displayLimit);
+        }
+      },
+      error(msg) {
+        console.log('Error is getting course fetching: ', msg);
+      }
+    });
   }
 
   ngOnInit(): void {
     this.hasMore = this.coursesService.isNotEmpty();
   }
 
   onLoadMore(): void {
-    console.log('Load more button was clicked');
+    this.courses = this.coursesService.getAll(++this.page, this.displayLimit);
```

Comment:
perfectly, it would be nice to have like this without using any additional state properties:
```typescript
this.courses$ = this.coursesService.getAll(++this.page, this.displayLimit);
```

Reply:
changed in next PR and skipped avoiding conflicts